code/aggregation/aggregate-raw-data.R

```diff
@@ -215,85 +207,103 @@ checkDuplicateFeatures <- function(df) {
+spa.data <- merge(spa.data, spa.ipa, all.x=TRUE, sort=FALSE)
+spa.data$spaAllophoneDescription <- removeBrackets(spa.data$spaAllophoneDescription)
+spa.data$Allophones <- spa.ipa$Phoneme[match(spa.data$spaAllophoneDescription,
+                                             spa.ipa$spaDescription)]
+## SPA does not include marginal phonemes; we denote this with NA
+spa.data$Marginal <- NA
+## clean up
+spa.data <- cleanUp(spa.data, "spa")
 rm(spa.raw, spa.ipa, spa.iso)
-## TODO: add inventory ID for SPA
-
-## UPSID
-upsid.ipa <- read.delim(upsid.ipa.path, na.strings="", quote="",
-                        stringsAsFactors=FALSE)
-upsid.segments <- read.delim(upsid.segments.path, na.strings="", quote="",
-                             stringsAsFactors=FALSE)
-upsid.language.codes <- read.delim(upsid.language.codes.path, na.strings="",
-                                   quote="", stringsAsFactors=FALSE)
+
+## UPSID has database-like tables, so we basically just merge things
+upsid.language.codes <- read.delim(upsid.language.codes.path, na.strings="", quote="")
+upsid.segments <- read.delim(upsid.segments.path, na.strings="", quote="")
+upsid.ipa <- read.delim(upsid.ipa.path, na.strings="", quote="")
+upsid.ipa <- upsid.ipa[c("upsidCCID", "Phoneme")]
 upsid.data <- merge(upsid.language.codes, upsid.segments, by="upsidLangNum")
-upsid.data <- merge(upsid.data, upsid.ipa[c("upsidCCID", "Phoneme")], by="upsidCCID")
-upsid.data <- within(upsid.data, {
-    Source <- "upsid"
-    Phoneme <- denorm(Phoneme)
-})
+upsid.data <- merge(upsid.data, upsid.ipa, by="upsidCCID")
+## upsidLangNum is already a unique (and meaningful) integer
+upsid.data$InventoryID <- upsid.data$upsidLangNum
+## add column for marginal phonemes
+upsid.data$Marginal <- as.logical(upsid.data$anomalous)
+## clean up
+upsid.data <- cleanUp(upsid.data, "upsid")
 rm(upsid.ipa, upsid.segments, upsid.language.codes)
-## TODO: add inventory ID for UPSID
 
-## RAMASWAMI
+## RAMASWAMI is a wide-format data source: 1 row per language, phonemes as
+## column headers, with boolean presence/absence indicators in the cells.
+## There is an integer ID in the first column of the raw data.
 ra.raw <- read.delim(ra.path, na.strings="", quote="", as.is=TRUE, header=FALSE)
-ra.data <- apply(ra.raw[4:nrow(ra.raw),], 1,
-				 function(i) data.frame(LanguageName=i[2], LanguageCode=i[3], InventoryID=i[1],
-				 Phoneme=c(t(ra.raw[2, 4:length(i)][as.logical(as.numeric(i[4:length(i)]))])),
-				 row.names=NULL))
+ra.data <- apply(ra.raw[4:nrow(ra.raw),], 1, function (i)
+    data.frame(InventoryID=i[1], LanguageName=i[2], LanguageCode=i[3],
+               Phoneme=c(t(ra.raw[2, 4:length(i)][as.logical(as.numeric(i[4:length(i)]))])),
+               row.names=NULL))
 ra.data <- do.call(rbind, ra.data)
-ra.data$Source <- "ra"
+## RA does not indicate marginal phonemes; we denote this with NA
+ra.data$Marginal <- NA
+## clean up
+ra.data <- cleanUp(ra.data, "ra")
 rm(ra.raw)
 
-## SAPHON
```

Comment:
the SAPHON code has changed very little; mostly just better naming and comments.